runtime/src/bank.rs

```diff
@@ -456,6 +456,18 @@ impl Bank {
             {
                 entered_epoch_callback(&mut new)
```

Comment:
This trick works for native programs but not built-in programs, so we need the new mechanism below for builtins